core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -34,7 +34,7 @@ object Json {
    * Parse a JSON string into a JsonValue if possible. `None` is returned if `input` is not valid JSON.
    */
   def parseFull(input: String): Option[JsonValue] =
-    try Option(mapper.readTree(input)).map(JsonValue(_))
+    try doParseFull(input)
     catch { case _: JsonProcessingException => None }
```

Comment:
We can avoid the introduction of `doParseFull` by using `tryParseFull` in this method. Something like:
`tryParseFull(input).toOption`. We can do the same for the other methods that return `Option`.